include/AST/Decl/ParamDecl.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef GLU_AST_DECL_PARAMDECL_HPP
+#define GLU_AST_DECL_PARAMDECL_HPP
+
+#include "Decl/VarLetDecl.hpp"
+
+namespace glu::ast {
+
+/// @class ParamDecl
+/// @brief Represents a parameter in a function declaration.
+///
+/// This class inherits from VarLetDecl and encapsulates the details of a
+/// function parameters declaration.
+class ParamDecl : public VarLetDecl {
+public:
+    /// @brief Constructor for the ParamDecl class.
+    /// @param location The source location of the declaration.
+    /// @param name The name of the declared param.
+    /// @param type The type of the declared param.
+    /// @param value The value assigned to the declared param.
+    ParamDecl(
+        SourceLocation location, std::string name, glu::types::TypeBase *type,
```

Comment:
note that VarLetDecl should probably be taking StringRef as well. we shouldn't have any heap-allocating members in our AST, that only causes leaks.